api/envoy/extensions/wasm/v3/wasm.proto

```diff
@@ -18,6 +18,33 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // [#protodoc-title: Wasm]
 // [#extension: envoy.bootstrap.wasm]
 
+// Configuration for restricting proxy-wasm capabilities available to modules.
+message CapabilityRestrictionConfig {
+  // The proxy-wasm capabilities which will be allowed. Capabilities are mapped by
+  // name. The *SanitizationConfig* which each capability maps to is currently unimplemented and ignored,
+  // and so should be left empty.
+  //
+  // The capability names are given in the
+  // `proxy-wasm ABI <https://github.com/proxy-wasm/spec/tree/master/abi-versions/vNEXT>`_.
+  // Additionally, the following WASI capabilities from
+  // `this list <https://github.com/WebAssembly/WASI/blob/master/phases/snapshot/docs.md#modules>`_
+  // are implemented and can be allowed:
+  // *fd_write*, *fd_read*, *fd_seek*, *fd_close*, *fd_fdstat_get*, *environ_get*, *environ_sizes_get*,
```

Comment:
everything listed here except fd_write is not supported actually as you can see, for instance, https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/master/src/exports.cc#L727-L732 .

Also clock_time_get and get_random are implemented now https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/master/src/exports.cc#L804-L829

And in the future, we may support get_environ since there's a feature request in the multiple SDK repositories, https://github.com/proxy-wasm/spec/issues/19.

Reply:
Even though the host implementation of WASI is indeed a bunch of mostly non-working (but adhering to spec) stubs, modules compiled against WASI want to import them, and this PR is purely about allowing/rejecting such calls.